game-core/src/test/java/games/strategy/triplea/util/MovableUnitsFilterTest.java

```diff
@@ -0,0 +1,115 @@
+package games.strategy.triplea.util;
+
+import static games.strategy.triplea.delegate.GameDataTestUtil.armour;
+import static games.strategy.triplea.delegate.GameDataTestUtil.germans;
+import static games.strategy.triplea.delegate.GameDataTestUtil.infantry;
+import static games.strategy.triplea.delegate.GameDataTestUtil.territory;
+import static games.strategy.triplea.delegate.MockDelegateBridge.advanceToStep;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsInAnyOrder;
+import static org.hamcrest.Matchers.empty;
+import static org.hamcrest.collection.IsCollectionWithSize.hasSize;
+import static org.hamcrest.core.IsNull.nullValue;
+
+import games.strategy.engine.data.GameData;
+import games.strategy.engine.data.PlayerId;
+import games.strategy.engine.data.Route;
+import games.strategy.engine.data.Territory;
+import games.strategy.engine.data.Unit;
+import games.strategy.engine.data.UnitType;
+import games.strategy.engine.delegate.IDelegateBridge;
+import games.strategy.triplea.delegate.AbstractMoveDelegate;
+import games.strategy.triplea.delegate.Matches;
+import games.strategy.triplea.delegate.MockDelegateBridge;
+import games.strategy.triplea.xml.TestMapGameData;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Predicate;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.triplea.java.collections.CollectionUtils;
+
+public class MovableUnitsFilterTest {
+  final GameData gameData = TestMapGameData.REVISED.getGameData();
```

Comment:
nit, these variables can be declared static: https://github.com/triplea-game/triplea/wiki/Java-Code-Conventions#mark-values-and-methods-as-static-where-possible-5489

Reply:
I think this throws, which I'm not sure it can be static? See my other comment - maybe we make it throw RuntimeException to avoid this?

Reply:
I agree with that direction. It's not great we have to use full-blown XML maps IMO to do testing, for now though it's certainly an improvement and helps enable cleaner tests.